server/API/REST.php

```diff
@@ -537,6 +572,7 @@ function set_lecture_online_status($vars) {
 	$r->addRoute('POST', API_PATH . '/logout', ['do_logout', NEED_AUTH]);
 	$r->addRoute('GET', API_PATH . "/types", "print_types");
 
+	$r->addRoute('GET', API_PATH . "/courses", "print_courses");
```

Comment:
Unauthenticated users should not be able to get the list of courses
```suggestion
	$r->addRoute('GET', API_PATH . "/courses", ["print_courses", NEED_AUTH]);
```

Reply:
no, is not actually needed, because there is a situation in which you don't need authentication (only list the courses)

Reply:
Which story is it?

Reply:
the story itself does not specify that, but is something that might be useful in the future, and there is no actual problem in make the course list available

Reply:
- There is a situation
- It may be useful in the future
Which one is it, and in which context?

Reply:
for example if you need to get the list of the courses without any filter (for statistics or listing)